Formula/s/single-file.rb

```diff
@@ -0,0 +1,20 @@
+require "language/node"
+
+class SingleFile < Formula
+  desc "CLI tool for saving a faithful copy of a complete web page in a single HTML file"
+  homepage "https://github.com/gildas-lormeau/single-file-cli/"
+  url "https://github.com/gildas-lormeau/single-file-cli/archive/refs/tags/v2.0.29.tar.gz"
+  sha256 "773727768135fe789476b2b3828704f3c44eb562147cece5b590a275706d9447"
+  license "AGPL-3.0-or-later"
+
+  depends_on "node"
+
+  def install
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/single-file --version")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Yeah, we definitely don't want tests installing additional things so the failure check seems fine.